backend/apiPackage.ts

```diff
@@ -67,4 +69,31 @@ export async function getPackagesByName(req : Request, res: Response){
   }catch(error){
     return res.status(500).send(`Error in getPackagesByName: ${error}`);
   }
-}
\ No newline at end of file
+}
+
+export async function extractMetadataFromZip(filebuffer: Buffer): Promise<apiSchema.PackageMetadata> {
+  const zip = await JSZip.loadAsync(filebuffer);
+  
+  let packageFile = zip.file("package.json");
```

Comment:
On package upload, either Content or URL should be set. If both are set, returns 400.
On package update, exactly one field should be set.
On download, the Content field should be set.

So you need to check for this. And you need to check that the PackageData was passed in with the request.